### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by CheshireCaat**
This PR contains some commits that contains code refactoring for:

TeleSharp.Generator:
- Removed unused references in Program.cs and replace 'String' to 'string'
- Added JsonProperty attribute for each property in models
- Models moved to separate folders

TLSharp.Core.MTProto.Serializers:
- Fix naming for methods and remove unused method

TLSharp.Core.DataCenter:
- Remove redundant setters for properties

### CODE REVIEW

File: TLSharp.Core/DataCenter.cs [hunk 1]
```diff
@@ -1,21 +1,20 @@
-
 namespace TLSharp.Core
 {
     internal class DataCenter
     {
-        internal DataCenter (int? dcId, string address, int port)
+        internal DataCenter(int? dcId, string address, int port)
```
Comment (knocte) [inline TLSharp.Core/DataCenter.cs]: no need to change this line above

File: TLSharp.Core/DataCenter.cs [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
         {
             DataCenterId = dcId;
             Address = address;
             Port = port;
         }
 
-        internal DataCenter (string address, int port) : this (null, address, port)
+        internal DataCenter(string address, int port) : this(null, address, port)
```
Comment (knocte) [inline TLSharp.Core/DataCenter.cs]: same here

Comment (CheshireCaat) [reply TLSharp.Core/DataCenter.cs]: Fixed it

Comment (knocte) [reply TLSharp.Core/DataCenter.cs]: can you squash the 2nd commit with the 4th commit?

Comment (CheshireCaat) [reply TLSharp.Core/DataCenter.cs]: Done

**Comment by knocte** [comment]
I guess you tested TLSharp after these changes?

**Comment by CheshireCaat** [comment]
Yes, i always check that library working fine using small console-client. Should I test specific functions? I guess that this PR not contains changes that may be break functionality of TLSharp.

Only thing that TeleSharp.Generator wasn't tested by me.

**Comment by knocte** [comment]
Well, the generator is the most important thing to test here because you used new Json attributes.

**Comment by CheshireCaat** [comment]
Ok, i tested it now with 23 layer from official docs. It works without any exceptions.

**Pull request merged**